### PR TITLE
s/documentation/language reference/ , also did minor grammar tweaks

### PR DESCRIPTION
**Pull request by WildYorkies**

### CODE REVIEW

File: syntax_and_semantics/return_types.md [hunk 1]
```diff
@@ -20,11 +20,13 @@ The return type follows the [type grammar](type_grammar.html).
 Marking a method as returning `Nil` will make it return `nil` regardless of what it actually returns:
 
 ```crystal
```
Comment (mverzilli) [inline syntax_and_semantics/return_types.md]: This change seems a bit unrelated to the goal of the PR and issue #78. What's the rationale for introducing this change here?

File: using_the_compiler/README.md [hunk 2]
```diff
@@ -12,14 +12,14 @@ To compile and run a program in a single shot you can invoke `crystal` with a si
 $ crystal some_program.cr
 ```
 
-Crystal files end with the `.cr` extension.
```
Comment (mverzilli) [inline using_the_compiler/README.md]: I agree that this reads weird, but I'm not sure the change fixes that. What do you think about moving the "Crystal files end with the `.cr` extension" sentence a bit upwards so that the previous sentence reads:

"To compile and run a program in a single shot, you can invoke `crystal` with a single filename (Crystal files end with `.cr`): (...)"

File: syntax_and_semantics/return_types.md (hunk 1 above)
Comment (WildYorkies) [inline syntax_and_semantics/return_types.md]: I don't see a change anywhere. Which line?

File: syntax_and_semantics/return_types.md [hunk 3]
```diff
@@ -20,11 +20,13 @@ The return type follows the [type grammar](type_grammar.html).
...
-def some_method : Nil
-  1 + 2
+def greet(name : String) : Nil
```
Comment (mverzilli) [inline syntax_and_semantics/return_types.md]: > @mverzilli I don't see a change anywhere. Which line?

Sorry, I meant here :)

File: syntax_and_semantics/literals/string.md [hunk 4]
```diff
@@ -145,9 +145,9 @@ b = 2
 
 This ends up invoking `Object#to_s(IO)` on each expression enclosed by `#{...}`.
 
-## Without interpolation nor escapes
+## Without interpolation or escapes
 
-To create a String without interpolation nor escapes use `%q`:
+To create a String without interpolation and without escapes use `%q`:
```
Comment (RX14) [inline syntax_and_semantics/literals/string.md]: ``To create a String without interpolation or escapes use `%q` `` reads better to me.

File: using_the_compiler/README.md [hunk 5]
```diff
@@ -6,20 +6,20 @@ In the next sections a dollar sign (`$`) denotes the command line.
...
+Crystal files end with the `.cr` extension.
+
 To compile and run a program in a single shot you can invoke `crystal` with a single filename:
 
 ```
 $ crystal some_program.cr
 ```
 
-Alternatively you can use the `run` command:
+Alternatively, you can use the `run` command and it will do the same thing:
```
Comment (RX14) [inline using_the_compiler/README.md]: Perhaps it would be better to introduce `crystal run ` first and then the short form?

Comment (mverzilli) [inline using_the_compiler/README.md]: Not sure about that. I think it's better to introduce the most minimalist form as the "default", and then clarify that actually ends up running the `run` command.

**Comment by mverzilli** [comment]
Next time I get to my computer I'll push the changes to his branch and merge, given we haven't heard from him for a couple of months.

**Comment by mverzilli** [comment]
@RX14 it took me more than expected to "get to my computer" 😅, but we can now merge this

**Comment by straight-shoota** [comment]
@mverzilli @RX14 Is this going to get merged for real some time? =)

**Comment by mverzilli** [comment]
Sure, we just need a second approval. @RX14?

**Comment by RX14** [comment]
@mverzilli I still see a bunch of unresolved reviews: specifically the removal of the code example change and the stuff around the `.cr` extension.

In addition, this PR has merge conflicts anyway.

**Comment by mverzilli** [comment]
I just sent a couple of PRs which cover this one and separate concerns: https://github.com/crystal-lang/crystal-book/pull/171 and https://github.com/crystal-lang/crystal-book/pull/172. Closing this one. Thanks @WildYorkies for taking the first stab!
